metadata-ingestion/src/datahub/ingestion/source/sql/trino.py

```diff
@@ -242,33 +251,35 @@ def _get_source_dataset_urn(
         table: str,
     ) -> Optional[str]:
         catalog_name = dataset_name.split(".")[0]
-        connector_name = get_catalog_connector_name(catalog_name, inspector)
-        if connector_name:
-            connector_platform_name = KNOWN_CONNECTOR_PLATFORM_MAPPING.get(
-                connector_name
+        connector_name = get_catalog_connector_name(inspector.bind, catalog_name)
+        if not connector_name:
+            return None
+        connector_details = self.config.catalog_to_connector_details.get(
+            catalog_name, ConnectorDetail()
+        )
+        connector_platform_name = KNOWN_CONNECTOR_PLATFORM_MAPPING.get(
+            connector_details.connector_platform
+            if connector_details.connector_platform
+            else connector_name
+        )
+        if not connector_platform_name:
+            logging.debug(f"Platform '{connector_platform_name}' is not yet supported.")
+            return None
+
+        if connector_platform_name in TWO_TIER_CONNECTORS:  # connector is two tier
+            return make_dataset_urn_with_platform_instance(
+                platform=connector_platform_name,
+                name=f"{schema}.{table}",
+                platform_instance=connector_details.platform_instance,
+                env=connector_details.env,
+            )
+        elif connector_details.connector_database:  # else connector is three tier
```

Comment:
add an else clause that reports a warning that the connector_database is missing